plugins/developer_guide.md

```diff
@@ -0,0 +1,305 @@
+# Guide to Dokka Plugin development
+
+## Configuration
+
+Dokka requires configured kotlin plugin and dokka-core dependency. 
+
+```kotlin
+plugins {
+    kotlin("jvm") version <kotlin_version>
+}
+
+dependencies {
+    implementation("org.jetbrains.dokka:dokka-core:<dokka_version>>)
+}
+
+tasks.withType<KotlinCompile> {
+    kotlinOptions.jvmTarget = "1.8"
+}
+```
+
+## Building sample plugin
+
+In order to load a plugin into dokka, your class must extend `DokkaPlugin` class. All instances are automatically 
+loaded during dokka setup using `ServiceLoader`.
```

Comment:
it might be worth to mention that `ServiceLoader` is from `java.util` and it not a custom implementation. I'd change `ServiceLoader` to `java.util.ServiceLoader`